tempodb/encoding/vparquet3/schema.go

```diff
@@ -473,6 +480,28 @@ func traceToParquet(meta *backend.BlockMeta, id common.ID, tr *tempopb.Trace, ot
 		}
 	}
 
+	// Calculate service meta information/statistics per trace
+	ot.ServicesMeta = []ServiceMeta{}
```

Comment:
This creates an entry for every batch.  I think we need to dedupe and sum.

Reply:
The same `res.Resource.ServiceName` can appear multiple times in this loop? I guess then I can create a map indexed by the service name and convert it to a list later?

Or keep the map in the block schema. I'm not sure if a list or map is better there - I assumed lists are easier to handle with parquet. Purely from the data perspective, a map would make more sense, as this is a grouping of per-service meta information.

Reply:
Yes, it's typical to receive many batches for the same resource because of the practical upper-limit on the batch size. The agent/collector can only buffer a limited amount of spans or length of time before it has to forward the batch onward.  

Map vs list:  they both end up in the parquet file nearly identical, key column + data columns, the only difference is in the parquet metadata (afaik, we haven't used MAP columns before).  And the low-level logic in Fetch() to return the data will be the same.  So I don't think I have strong opinions either way.    A map may be simpler because it will be allocated anyway.

Reply:
I've updated the PR to sum up the service statistics and use a map instead. As you mentioned it's actually quite straightforward to convert the list to a map, as the data is stored similarly :)